frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -297,6 +306,47 @@ end
 
 function ReaderHighlight:onReaderReady()
     self:setupTouchZones()
+    self:migrateHighlights()
+end
+
+
+function ReaderHighlight:migrateHighlights()
+    if self.ui.doc_settings:hasNot("highlight_merged") then
+        for page, highlights in pairs(self.view.highlight.saved) do
+            for _, item in ipairs(highlights) do
+                -- This is to switch notes and text.
+                local highlight = bookmarkToLocalHighlight(item)
+                if not highlight.chapter then
+                    local pg_or_xp = self.ui.paging and page or highlight.pos0
+                    local chapter_name = self.ui.toc:getTocTitleByPage(pg_or_xp)
+                    highlight.chapter = chapter_name
```

Comment:
What are the reasons to introduce `local chapter_name` not used anymore?

Reply:
fixing it.

Reply:
Done. Also, I'm grateful that you're reviewing this despite the fact that you were (probably still are) opposed to this refactor. :)